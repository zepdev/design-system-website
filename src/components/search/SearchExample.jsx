import React from 'react'
import Search from '@zlab-de/zel-react/Search'

function SearchExample() {
  const handleChange = selection => {
    alert(`You selected ${selection.value}`)
  }
  const searchItems = [
    { value: 'value 1' },
    { value: 'value 2' },
    { value: 'value 3' },
  ]

  return (
    <div>
      <Search
        onChange={handleChange}
        items={searchItems}
        placeholder="Search for value"
      />

      <Search
        onChange={handleChange}
        items={searchItems}
        variant="header"
        placeholder="Search for value"
      />
    </div>
  )
}

export default SearchExample
