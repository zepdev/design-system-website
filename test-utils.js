import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
