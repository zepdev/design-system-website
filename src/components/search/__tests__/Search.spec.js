import React from 'react'
import { render } from '../../../../test-utils'
import SearchExample from '../SearchExample'

describe('SearchExample', () => {
  it('renders correctly', () => {
    const component = render(<SearchExample />)
    expect(component).toMatchSnapshot()
  })
})
