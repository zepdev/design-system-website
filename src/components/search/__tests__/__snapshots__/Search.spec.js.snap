// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            class="zep-search"
            role="combobox"
          >
            <label
              class="zep-visually-hidden"
              for="downshift-0-input"
              id="downshift-0-label"
            >
              Search for value
            </label>
            <input
              aria-autocomplete="list"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="zep-search__input zep-search__input--rounded-left"
              id="downshift-0-input"
              placeholder="Search for value"
              value=""
            />
            <svg
              class="zep-search__icon"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                zepicons-search
              </title>
              <path
                d="M12.667 4c4.786 0 8.667 3.88 8.667 8.667 0 2.147-0.787 4.12-2.080 5.64l0.36 0.36h1.053l6.667 6.667-2 2-6.667-6.667v-1.053l-0.36-0.36c-1.52 1.293-3.493 2.080-5.64 2.080-4.786 0-8.667-3.88-8.667-8.667s3.88-8.667 8.667-8.667v0zM12.667 6.667c-3.333 0-6 2.667-6 6s2.667 6 6 6c3.333 0 6-2.667 6-6s-2.667-6-6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          role="search"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-1-label"
            class="zep-search"
            role="combobox"
          >
            <label
              class="zep-visually-hidden"
              for="downshift-1-input"
              id="downshift-1-label"
            >
              Search for value
            </label>
            <input
              aria-autocomplete="list"
              aria-labelledby="downshift-1-label"
              autocomplete="off"
              class="zep-search__input zep-search__input--searchfield"
              id="downshift-1-input"
              placeholder="Search for value"
              value=""
            />
            <svg
              class="zep-search__icon"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                zepicons-search
              </title>
              <path
                d="M12.667 4c4.786 0 8.667 3.88 8.667 8.667 0 2.147-0.787 4.12-2.080 5.64l0.36 0.36h1.053l6.667 6.667-2 2-6.667-6.667v-1.053l-0.36-0.36c-1.52 1.293-3.493 2.080-5.64 2.080-4.786 0-8.667-3.88-8.667-8.667s3.88-8.667 8.667-8.667v0zM12.667 6.667c-3.333 0-6 2.667-6 6s2.667 6 6 6c3.333 0 6-2.667 6-6s-2.667-6-6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="downshift-0-label"
          class="zep-search"
          role="combobox"
        >
          <label
            class="zep-visually-hidden"
            for="downshift-0-input"
            id="downshift-0-label"
          >
            Search for value
          </label>
          <input
            aria-autocomplete="list"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class="zep-search__input zep-search__input--rounded-left"
            id="downshift-0-input"
            placeholder="Search for value"
            value=""
          />
          <svg
            class="zep-search__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              zepicons-search
            </title>
            <path
              d="M12.667 4c4.786 0 8.667 3.88 8.667 8.667 0 2.147-0.787 4.12-2.080 5.64l0.36 0.36h1.053l6.667 6.667-2 2-6.667-6.667v-1.053l-0.36-0.36c-1.52 1.293-3.493 2.080-5.64 2.080-4.786 0-8.667-3.88-8.667-8.667s3.88-8.667 8.667-8.667v0zM12.667 6.667c-3.333 0-6 2.667-6 6s2.667 6 6 6c3.333 0 6-2.667 6-6s-2.667-6-6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        role="search"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="downshift-1-label"
          class="zep-search"
          role="combobox"
        >
          <label
            class="zep-visually-hidden"
            for="downshift-1-input"
            id="downshift-1-label"
          >
            Search for value
          </label>
          <input
            aria-autocomplete="list"
            aria-labelledby="downshift-1-label"
            autocomplete="off"
            class="zep-search__input zep-search__input--searchfield"
            id="downshift-1-input"
            placeholder="Search for value"
            value=""
          />
          <svg
            class="zep-search__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              zepicons-search
            </title>
            <path
              d="M12.667 4c4.786 0 8.667 3.88 8.667 8.667 0 2.147-0.787 4.12-2.080 5.64l0.36 0.36h1.053l6.667 6.667-2 2-6.667-6.667v-1.053l-0.36-0.36c-1.52 1.293-3.493 2.080-5.64 2.080-4.786 0-8.667-3.88-8.667-8.667s3.88-8.667 8.667-8.667v0zM12.667 6.667c-3.333 0-6 2.667-6 6s2.667 6 6 6c3.333 0 6-2.667 6-6s-2.667-6-6-6z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
